veggieseasons/lib/screens/details.dart

```diff
@@ -4,18 +4,156 @@
 
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+import 'package:scoped_model/scoped_model.dart';
+import 'package:veggieseasons/data/model.dart';
+import 'package:veggieseasons/data/veggie.dart';
 import 'package:veggieseasons/styles.dart';
 
+class SeasonCircle extends StatelessWidget {
+  final Season season;
+
+  SeasonCircle(this.season);
+
+  String get _firstChars {
+    return '${season.toString().substring(7, 8).toUpperCase()}'
+        '${season.toString().substring(8, 9)}';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(4.0),
+      child: DecoratedBox(
+        decoration: BoxDecoration(
+          color: Styles.seasonColors[season],
+          borderRadius: BorderRadius.circular(25.0),
+          border: Styles.seasonBorder,
+        ),
+        child: SizedBox(
+          height: 50.0,
+          width: 50.0,
+          child: Center(child: Text(_firstChars, style: Styles.seasonText)),
+        ),
+      ),
+    );
+  }
+}
+
 class DetailsScreen extends StatelessWidget {
+  final int id;
+
+  DetailsScreen({@required this.id});
```

Comment:
personally, having a single, required named argument feels a little odd, but I know this is often the Flutter style. Because it's just one argument, and the argument is "id", I'm inclined to just have one positional argument, but up to you.

Reply:
That's valid. Changed to a positional argument.

I think some earlier draft had another argument in there, and it's been removed.